src/content/docs/extensions/vector.mdx

```diff
@@ -0,0 +1,270 @@
+---
+title: "Vector search"
+---
+
+The `VECTOR` extension provides an on-disk [HNSW](https://en.wikipedia.org/wiki/Hierarchical_navigable_small_world)-based vector index for accelerating similarity search over float array columns in tables.
+
+This feature has some limitations for now, but will evolve over time:
+
+- **Immutable Index**: Once created, the index cannot be modified. You must drop and re-create the index to reflect changes in the underlying tables.
+- **Float Array Support Only**: Currently only supports float array columns. Support for additional data types is planned for future releases.
+- **Single Node Table Column Indexing**: Limited to indexing a single column in node tables.
+
+The vector extension provides the following functions:
+| Function | Description |
+| --- | --- |
+| `CREATE_VECTOR_INDEX` | Create the index |
+| `QUERY_VECTOR_INDEX` | Query the index |
+| `DROP_VECTOR_INDEX` | Drop the index |
+
+
+## Installation
+To get started with the vector index extension, you need to first install and load the extension following the [Using Extensions in Kuzu](/extensions#using-extensions-in-kuzu) instructions.
+
+```sql
+INSTALL VECTOR;
+LOAD VECTOR;
+```
+
+## Basic Usage
+Below is an example demonstrating how to create and use a vector index on a `Book` table. In this example, we'll create 384-dimensional embedding vectors for book titles using the `sentence-transformers` library.
+
+In the following example, we'll generate embeddings using the `sentence-transformers` library in Python:
+```python
+# create_embeddings.py
+# pip install sentence-transformers
+from sentence_transformers import SentenceTransformer
+
+# Load a pre-trained embedding generation model
+# https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2
+model = SentenceTransformer("all-MiniLM-L6-v2")
+
+# Generate embeddings for sample book titles
+titles = [
+    "The Quantum World",
+    "Chronicles of the Universe",
+    "Learning Machines",
+    "Echoes of the Past",
+    "The Dragon's Call"
+]
+
+for title in titles:
+    # Convert title to a 384-dimensional embedding vector
+    embeddings = model.encode(title).tolist()
+    print(embeddings)
+```
+
+The printed embeddings are only for display purposes. In the next step, we will insert these embeddings
+as a node property in Kuzu.
+
+### Example dataset
+
+Let's create a `Book` node table with a `title_embedding` column to store our 384-dimensional embeddings.
+In our DDL, we define this property as an array of floats with 384 dimensions.
+
+```sql
```

Comment:
We should probably continue the example with python instead of Cypher. Writing these vectors manually looks bad.